src/build/gen/public_utils.ts

```diff
@@ -0,0 +1,58 @@
+import { Project } from "../../project/mod.ts";
+import { GeneratedFile } from "./mod.ts";
+import { resolve } from "../../deps.ts";
+import { camelify } from "../../types/case_conversions.ts";
+
+export const UTILS_GEN_FILE = "public.ts";
+
+export async function compilePublicUtilsHelpers(project: Project) {
+	const projectUtilsGenFolder = resolve(
+		project.path,
+		"_gen",
+		"public_utils",
+	);
+
+	const allPublicUtils = new GeneratedFile(resolve(projectUtilsGenFolder, "all.ts"));
+
+	for (const module of project.modules.values()) {
+		const reexporter = new GeneratedFile(resolve(
+			projectUtilsGenFolder,
+			`mod_${module.name}.ts`, // In the off-chance there is a module called "all"
+		), 2);
+
+		// Find the path to the `public.ts` file for this module
+		const modulePublicUtilsPath = resolve(
+			module.path,
+			UTILS_GEN_FILE,
+		);
+
+		// Check if the module has a public utils file
+		const modulePublicUtilsExists = await Deno.stat(modulePublicUtilsPath).then(() => true, () => false);
+
+		if (modulePublicUtilsExists) {
```

Comment:
```suggestion
		if (await exists(modulePublicUtilsPath, { isFile: true })) {
```

Reply:
At least it's kind of poetic that I forgot that `exists` exists?